monai/apps/detection/utils/anchor_utils.py

```diff
@@ -0,0 +1,406 @@
+                shifts_centers[axis] = shifts_centers[axis].reshape(-1)
+
+            # Expand to [x_center, y_center, x_center, y_center],
+            # or [x_center, y_center, z_center, x_center, y_center, z_center]
+            if self.indexing == "xy":
+                # Cartesian ('xy') indexing swaps axis 0 and 1.
+                shifts_centers[1], shifts_centers[0] = shifts_centers[0], shifts_centers[1]
+            shifts = torch.stack(shifts_centers * 2, dim=1)  # sized (HW,4) or (HWD,6)
+
+            # For every (base anchor, output anchor) pair,
+            # offset each zero-centered base anchor by the center of the output anchor.
+            anchors.append(
+                (shifts.view(-1, 1, self.spatial_dims * 2) + base_anchors.view(1, -1, self.spatial_dims * 2)).reshape(
+                    -1, self.spatial_dims * 2
+                )  # each element sized (AHWD,4) or (AHWD,6)
+            )
+
+        return anchors
+
+    def forward(self, images: Tensor, feature_maps: List[Tensor]) -> List[Tensor]:
+        """
+        Generate anchor boxes for each image.
+
+        Args:
+            images: sized (B, C, W, H) or (B, C, W, H, D)
+            feature_maps: for FPN level i, feature_maps[i] is sizec (B, C_i, W_i, H_i) or (B, C_i, W_i, H_i, D_i).
+                This input argument does not have to be the actual feature maps.
+                Any list variable with the same (C_i, W_i, H_i) or (C_i, W_i, H_i, D_i) as feature maps works.
+
+        Return:
+            A list with length of B. Each element represents the anchors for this image.
+            The B elements are identical.
+
+        Example:
+            .. code-block:: python
+
+                images = torch.zeros((3,1,128,128,128))
+                feature_maps = [torch.zeros((3,6,64,64,32)), torch.zeros((3,6,32,32,16))]
+        """
+        grid_sizes = [feature_map.shape[-self.spatial_dims :] for feature_map in feature_maps]
+        image_size = images.shape[-self.spatial_dims :]
+        batchsize = images.shape[0]
+        dtype, device = feature_maps[0].dtype, feature_maps[0].device
+        strides = [
+            [
+                torch.tensor(image_size[axis] // g[axis], dtype=torch.int64, device=device)
+                for axis in range(self.spatial_dims)
+            ]
+            for g in grid_sizes
+        ]
+
+        # Code below come from torchvision.models.detection.anchor_utils.AnchorGenerator.forward()
+        self.set_cell_anchors(dtype, device)
+        anchors_over_all_feature_maps = self.grid_anchors(grid_sizes, strides)
+
+        anchors: List[List[torch.Tensor]] = []
+        for _ in range(batchsize):
+            anchors_in_image = list(anchors_over_all_feature_maps)
+            anchors.append(anchors_in_image)
+        return [torch.cat(anchors_per_image) for anchors_per_image in anchors]
+
+
+class AnchorGeneratorWithAnchorShape(AnchorGenerator, nn.Module):
```

Comment:
this nn.module could be removed, please help double-check (is it here because of torchscript?)
```suggestion
class AnchorGeneratorWithAnchorShape(AnchorGenerator):
```